bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerMetadata.java

```diff
@@ -308,11 +319,13 @@ long getNextEnsembleChange(long entryId) {
      *            byte array to parse
      * @param version
      *            version of the ledger metadata
+     * @param zkCtime
+     *            ctime read from Zookeeper Stat, used for legacy ledgers
      * @return LedgerConfig
      * @throws IOException
      *             if the given byte[] cannot be parsed
      */
-    public static LedgerMetadata parseConfig(byte[] bytes, Version version) throws IOException {
+    public static LedgerMetadata parseConfig(byte[] bytes, Version version, Long zkCtime) throws IOException {
```

Comment:
change 'zkCtime' to 'msCtime' (stands for metadata store creation time)? since the ledger metadata here isn't actually tight with zookeeper time.

And I'd prefer Optional<Long> rather than Long.